neuralpp/symbolic/z3_expression.py

```diff
@@ -406,7 +406,7 @@ def dict(self) -> Dict[str, Any]:
     def __eq__(self, other) -> bool:
         return False  # why do we need to compare two Z3ConjunctiveClause?
 
-    def __init__(self, z3_solver: z3.Solver, value_dict: Dict[str, Any] | None = None):
+    def __init__(self, z3_solver: z3.Solver, value_dict: Dict[str, Any] | None = None, unknown: bool = False):
```

Comment:
For this example, yes, `unknown` can just be checked by the function.

I don't think it should be checked lazily, since in `make()` we've already `check()`ed once.

Reply:
I found following comment a bit confusing.

> What if the Solver knows the satisfiability but the user says "unknown=True"?

I don't think this is a problem specific to this design. We can say this basically about any function interface that provides a "short cut" that aims to reduce computation overhead by getting some information from users but also puts some assumptions on the input. E.g., in this `__init__`, `value_dict` also serves this purpose: if it's supplied, we don't have to `_extract_key_value_from_assertions`, and users can (but should not) violate the contract by supplying a wrong `value_dict`, in which case, behavior of the function is undefined.